documents/anchor.go

```diff
@@ -18,6 +19,7 @@ type AnchorProcessor interface {
 	PreAnchorDocument(ctx context.Context, model Model) error
 	AnchorDocument(ctx context.Context, model Model) error
 	SendDocument(ctx context.Context, model Model) error
+	RequestDocumentWithToken(ctx context.Context, tokenIdentifier, entityIdentifier, entityRelationIdentifier []byte) (*p2ppb.GetDocumentResponse, error)
```

Comment:
```suggestion
	RequestDocumentWithAccessToken(ctx context.Context, tokenIdentifier, documentIdentifier, delegatingDocumentIdentifier []byte) (*p2ppb.GetDocumentResponse, error)
```
As a broader question, why do we need this definition in the `Anchor.go` file? I do not think has to do anything with `anchoring`. I would like to not expose any `p2ppb.*` here.
If you want to have a common method that is used by many other documents, I would create a new interface, maybe DocumentProcessor? or DocumentGetter? in a different file or under the processor.go that encapsulates the proto file calls.

Reply:
I agree. The processor was previously only used for anchoring but I would argue we can see it in a more generic way. For me, the processor communicates with the p2p layer.

`service -> processor -> p2p`

The method for transforming a `p2ppb.*` response to a model would be in the service.
I can't access the service in the processor it would create a cyclic dependency, that is the reason why I am returning the `p2ppb.*`

Reply:
Yes, that is fair. My suggestion is to move that definition to a different interface implemented by the defaultProcessor similar to `Send` that abstracts the p2ppb logic.